index.js

```diff
@@ -45,22 +45,27 @@ export const SKIP = 'skip'
  */
 export const EXIT = false
 
+/**
+ * Visit children of tree which pass a test
+ *
+ * @param tree Abstract syntax tree to walk
+ * @param test Test node, optional
+ * @param visitor Function to run for each node
+ * @param reverse Fisit the tree in reverse, defaults to false
```

Comment:
```suggestion
 * @param [reverse=false] Visit the tree in reverse order
```

Reply:
the `[]` brackets at this point don’t affect my editor (Atom). Do you see them having effect?

Reply:
:thinking: strange I don't see them in VSCode either, maybe because there is no `{boolean}`?
It could also be fine to keep the default in the description

```suggestion
 * @param reverse Visit the tree in reverse order, defaults to false
```

Reply:
I think the reason is because there are overloads, so values and defaults of things don’t matter (they differ per overload).

- The top-most one describes all overloads, so that’s why I don’t describe types and explicitly in prose write whether things are optional
- One level deeper is TS only. Docs are impossible.
- Level 3 is for the actual code and is used to check JS, so descriptions and prose don’t matter